src/stream/ngx_stream_proxy_module.c

```diff
@@ -1001,6 +1035,19 @@ ngx_stream_proxy_ssl_init_connection(ngx_stream_session_t *s)
 
     pscf = ngx_stream_get_module_srv_conf(s, ngx_stream_proxy_module);
 
+    if (!pscf->ssl_synced) {
+        if (ngx_ssl_certificate_modified(pscf->ssl, &pscf->ssl_certificate,
```

Comment:
Initially I wanted to do this inside event handler and just reuse `ngx_ssl_certificate` function. But I could not find any way to access `ngx_conf_t cf` variable to pass to `ngx_ssl_certificate` function as a first argument. When I created it as a property of server configuration object, and called `ngx_ssl_certificate` in the even handler, the worker was getting `SIGHOLD` and exiting. I did not go far with debugging and instead just modified `ngx_ssl_certificate` to not use `cf` variable.